challenges/literal-types/prompt.md

```diff
@@ -0,0 +1,104 @@
+## The Problem Literal Types Solve
+
+JavaScript, like most programming languages has a concept of "primitive" data types.
+
+Primitive data types are things like `string`, `boolean`, `number`, and `object`.
+
+> in JavaScript, arrays and functions are actually objects, but that's a topic for another time :)
+
+Let's say we've been baking chocolate chip cookies as a side-hussle and we wanna take things to the next level.  To get to industiral scales, we're gonna need to start writing down our recipes.
+
+We might make some variables like this:
+
+```js
+const name = 'chocolate chips';
+const inStock = true;
+const kilograms = 5;
+```
+
+Or, better yet, we might wrap them all in an object:
+
+```js
+const chocolateChips = {
+  name: 'chocolate chips',
+  inStock: true,
+  kilograms: 5,
+}
+```
+
+TypeScript has built-in _types_ that represent all of these primitive data types.
+
+```ts
+const name: string = 'chocolate chips';
+const inStock: boolean = true;
+const kilograms: number = 5;
+
+interface Ingredient {
+  name: string;
+  inStock: boolean;
+  kilograms: number;
+}
+
+const chocolateChips: Ingredient = {
+  name: 'chocolate chips',
+  inStock: true,
+  kilograms: 5,
+}
+```
+
+You can also create [type aliases](todo-needs-link) for primitive types:
+
+```ts
+type Name = string;
+type InStock = boolean;
+type Kilograms = number;
+
+interface Ingredient {
+  name: Name;
+  inStock: InStock;
+  kilograms: Kilograms;
+}
+```
+
+But there's a problem.  Actually there are lots of problems that literal types solve but let's just look at one of them.
+
+Let's say you have a function like this:
+
+```ts
+const needToOrderMore = (ingredient: Ingredient) => {
+  if (ingredient.kilograms < 100) {
+    return true;
+  }
+  return false;
+}
+```
+
+## Using Literal Types
+
+```ts
+type Name = 'chocolate chips';
```

Comment:
I've gone back and forth on that.  If you thought it was strange enough to even mention it then that's all the evidence that I need that I should go back to type instead of interface.  I did so in 3d07288a7fa462440ab85fcac63f19397b4591ed